src/controllers/routers/cases.js

```diff
@@ -0,0 +1,145 @@
+/*!
+ * OneSG API Server by TL Bootcamp#6 OneSG Team
+ * Copyright(c) 2020 TechLadies
+ * MIT Licensed
+ */
+
+'use strict';
+
+const {
+  DataError,
+  ForeignKeyViolationError,
+  ValidationError,
+} = require('objection');
+
+const { Case } = require('../../models');
+
+const {
+  errors: { BadRequest, InvalidInput },
+} = require('../../utils');
+
+/**
+ * Sanitize data from client.
+ * Call before an insert or an update.
+ * @param {Object} json - Unsanitised case
+ * @return {Object} cases - Sanitised case
+ */
+
+function sanitizedCase(json) {
+  const cases = json;
+  if (json.caseStatus) {
+    cases.caseStatus = json.caseStatus.toUpperCase().trim();
+  }
+  if (json.pointOfContact) {
+    cases.pointOfContact = json.pointOfContact.trim();
+  }
+  if (json.referenceStatus) {
+    cases.referenceStatus = json.referenceStatus.toUpperCase().trim();
+  }
+  if (json.casePendingReason) {
+    cases.casePendingReason = json.casePendingReason.trim();
+  }
+  if (typeof json.amountRequested === 'string') {
+    cases.amountRequested = parseFloat(json.amountRequested);
+  }
+  if (typeof json.amountGranted === 'string') {
+    cases.amountGranted = parseFloat(json.amountGranted);
+  }
+  if (json.documents) {
+    if (json.documents === null || json.documents === '') {
+      cases.documents = {};
+    }
+  }
+  if (json.refereeId) {
+    cases.refereeId = parseInt(json.refereeId, 10);
+  }
+  if (json.beneficiaryId) {
+    cases.beneficiaryId = parseInt(json.beneficiaryId, 10);
+  }
+  if (json.createdBy) {
+    cases.createdBy = parseInt(json.createdBy, 10);
+  }
+  if (json.updatedBy) {
+    cases.updatedBy = parseInt(json.updatedBy, 10);
+  }
+  return cases;
+}
+
+/**
+ * Retrieve all cases
+ * @param {Request} req
+ * @param {Response} res
+ */
+const getAll = async (req, res) => {
+  const cases = await Case.query().select();
+  return res.status(200).json({ cases });
+};
+
+/**
+ * Create new case
+ * @param {Request} req
+ * @param {Response} res
+ */
+const create = async (req, res, next) => {
+  const newCase = sanitizedCase(req.body);
+  try {
+    return await Case.transaction(async (trx) => {
```

Comment:
tested the transaction and the rollback works

Reply:
yay